src/org/labkey/test/tests/DataClassFolderExportImportTest.java

```diff
@@ -0,0 +1,351 @@
+    {
+        String subfolder = "simpleDataClassExportFolder";
+        String subfolderPath = getProjectName() + "/" + subfolder;
+        String testDataClass = "testData?<>*/Class";    // having the dataClass with non-file-legal chars in it is intentional,
+                                                        // the import/export processes will write temp-files with names derived from
+                                                        // the dataClass name.  This ensures that the sanitized name on export can
+                                                        // be successfully matched up with the intended dataClass on import
+        String importFolder = "simpleDataClassImportFolder";
+
+        // create a subfolder in the import destination project
+        _containerHelper.createSubfolder(IMPORT_PROJECT_NAME, importFolder);
+        // create a subfolder in the project to hold types we'll export in this test method
+        _containerHelper.createSubfolder(getProjectName(), subfolder);
+
+        DataClassDefinition testType = new DataClassDefinition(testDataClass).setFields(DataClassAPIHelper.dataClassTestFields());
+
+        TestDataGenerator testDgen = DataClassAPIHelper.createEmptyDataClass(subfolderPath, testType);
+
+        testDgen.addCustomRow(Map.of("Name", "class1", "intColumn", 1, "decimalColumn", 1.1, "stringColumn", "one"));
+        testDgen.addCustomRow(Map.of("Name", "class2", "intColumn", 2, "decimalColumn", 2.2, "stringColumn", "two"));
+        testDgen.addCustomRow(Map.of("Name", "class3", "intColumn", 3, "decimalColumn", 3.3, "stringColumn", "three"));
+        testDgen.addCustomRow(Map.of("Name", "class4", "intColumn", 4, "decimalColumn", 4.4, "stringColumn", "four"));
+        testDgen.addCustomRow(Map.of("Name", "class5", "intColumn", 5, "decimalColumn", 5.5, "stringColumn", "five"));
+        testDgen.insertRows();
+
+        Integer expectedRecordCount = testDgen.getRowCount();
+
+        PortalHelper portalHelper = new PortalHelper(this);
+        portalHelper.addWebPart("Data Classes");
+        portalHelper.addWebPart("Experiment Runs");
+
+        DataRegionTable sourceRunsTable = DataRegionTable.DataRegion(getDriver()).withName("Runs").waitFor();
+        List<String> runNames = sourceRunsTable.getColumnDataAsText("Name");
+
+        clickAndWait(Locator.linkWithText(testDataClass));
+        DataRegionTable sourceTable = DataRegionTable.DataRegion(getDriver()).withName("query").waitFor();
+        List<Map<String, String>> sourceRowData = new ArrayList<>();
+        for (int i=0; i<_attachments.size(); i++)
+        {
+            sourceTable.clickEditRow(i);
+            setFormElement(Locator.input("quf_attachmentColumn"), _attachments.get(i));
+            clickButton("Submit");
+        }
+        for (int i=0; i<expectedRecordCount; i++)
+        {
+            sourceRowData.add(sourceTable.getRowDataAsMap(i));
+        }
```

Comment:
Consider adding a method to `DataRegionTable` to throw the whole thing into a `Map` instead of have tests gather it row-by-row. Something like:
```suggestion
        sourceRowData = sourceTable.getTableData();
```

Reply:
I like it, done